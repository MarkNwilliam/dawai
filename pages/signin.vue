<template>
  <v-layout align-center justify-center column>
    <v-flex xs12>
      <v-card :loading="loading" flat outlined width="500px">
        <div class="pa-4">
          <v-img
            contain
            height="200"
            :aspect-ratio="16 / 9"
            src="/logo-1024-transparent.svg"
          ></v-img>
        </div>
        <v-form ref="form" v-model="valid" lazy-validation>
          <v-card-text class="pa-5">
            <v-text-field
              v-model="email"
              :rules="emailRules"
              type="email"
              label="Email"
              :prepend-inner-icon="mdiEmail"
              outlined
            />
            <v-text-field
              v-model="password"
              :rules="passwordRules"
              type="password"
              label="Password"
              :prepend-inner-icon="mdiLock"
              outlined
            />
            <nuxt-link to="/">
              Forgot password?
            </nuxt-link>
          </v-card-text>
          <v-card-actions class="px-5">
            <v-checkbox
              v-model="check"
              color="primary"
              label="Keep me signed in"
              :on-icon="mdiCheckboxMarked"
              :off-icon="mdiCheckboxBlankOutline"
            />
            <v-spacer />
            <v-btn
              color="primary"
              type="submit"
              large
              @click.stop.prevent="signin"
            >
              Sign In
              <v-icon right>
                {{ mdiLogin }}
              </v-icon>
            </v-btn>
          </v-card-actions>
        </v-form>
      </v-card>
    </v-flex>
  </v-layout>
</template>

<script>
import {
  mdiEmail,
  mdiLock,
  mdiClose,
  mdiCheckboxMarked,
  mdiCheckboxBlankOutline,
  mdiLogin
} from '@mdi/js'
export default {
  layout: () => 'none',
  // middleware: 'no-auth',
  head: () => ({
    title: 'Sign In'
  }),
  data: () => ({
    mdiEmail,
    mdiLock,
    mdiClose,
    mdiCheckboxMarked,
    mdiCheckboxBlankOutline,
    mdiLogin,
    check: true,
    valid: true,
    loading: false,
    email: '',
    password: '',
    emailRules: [
      (v) => !!v || 'E-mail is required',
      (v) => /.+@.+\..+/.test(v) || 'E-mail must be valid'
    ],
    passwordRules: [(v) => !!v || 'Password is required']
  }),
  methods: {
    signin() {
      if (this.$refs.form.validate()) {
        this.loading = true
        setTimeout(() => {
          this.loading = false
          this.$router.push('/')
        }, 3000)
      }
    }
  }
}
</script>
