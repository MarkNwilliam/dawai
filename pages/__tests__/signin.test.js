// Libraries
import Vuetify from 'vuetify'

// Component
import { mount, createLocalVue } from '@vue/test-utils'
import signin from '../signin'

// Utilities
const localVue = createLocalVue()

describe('signin.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should render', done => {
    const wrapper = mount(signin, {
      localVue,
      vuetify
    })

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot()

    done()
  })
})