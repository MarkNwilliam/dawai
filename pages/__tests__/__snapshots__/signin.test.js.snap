// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signin.vue should render 1`] = `
"<v-layout align-center=\\"\\" justify-center=\\"\\" column=\\"\\"><v-flex xs12=\\"\\"><v-card flat=\\"\\" outlined=\\"\\" width=\\"500px\\" class=\\"slight-br\\"><div class=\\"pa-4\\"><v-img contain=\\"\\" height=\\"200\\" aspect-ratio=\\"1.7777777777777777\\" src=\\"/logo-1024-transparent.svg\\"></v-img></div> <v-form lazy-validation=\\"\\"><v-card-text class=\\"pa-5\\"><v-text-field rules=\\"v => {
          /* istanbul ignore next */
          cov_jmrxdhfxl.f[4]++;
          cov_jmrxdhfxl.s[8]++;
          return (
            /* istanbul ignore next */
            (cov_jmrxdhfxl.b[0][0]++, !!v) ||
            /* istanbul ignore next */
            (cov_jmrxdhfxl.b[0][1]++, 'E-mail is required')
          );
        },v => {
          /* istanbul ignore next */
          cov_jmrxdhfxl.f[5]++;
          cov_jmrxdhfxl.s[9]++;
          return (
            /* istanbul ignore next */
            (cov_jmrxdhfxl.b[1][0]++, /.+@.+\\\\..+/.test(v)) ||
            /* istanbul ignore next */
            (cov_jmrxdhfxl.b[1][1]++, 'E-mail must be valid')
          );
        }\\" type=\\"email\\" label=\\"Email\\" prepend-inner-icon=\\"M20,8L12,13L4,8V6L12,11L20,6M20,4H4C2.89,4 2,4.89 2,6V18A2,2 0 0,0 4,20H20A2,2 0 0,0 22,18V6C22,4.89 21.1,4 20,4Z\\" outlined=\\"\\" rounded=\\"\\"></v-text-field> <v-text-field rules=\\"v => {
          /* istanbul ignore next */
          cov_jmrxdhfxl.f[6]++;
          cov_jmrxdhfxl.s[10]++;
          return (
            /* istanbul ignore next */
            (cov_jmrxdhfxl.b[2][0]++, !!v) ||
            /* istanbul ignore next */
            (cov_jmrxdhfxl.b[2][1]++, 'Password is required')
          );
        }\\" type=\\"password\\" label=\\"Password\\" prepend-inner-icon=\\"M12,17A2,2 0 0,0 14,15C14,13.89 13.1,13 12,13A2,2 0 0,0 10,15A2,2 0 0,0 12,17M18,8A2,2 0 0,1 20,10V20A2,2 0 0,1 18,22H6A2,2 0 0,1 4,20V10C4,8.89 4.9,8 6,8H7V6A5,5 0 0,1 12,1A5,5 0 0,1 17,6V8H18M12,3A3,3 0 0,0 9,6V8H15V6A3,3 0 0,0 12,3Z\\" outlined=\\"\\" rounded=\\"\\"></v-text-field> <nuxt-link to=\\"/\\">
            Forgot password?
          </nuxt-link> <nuxt-link to=\\"/signup\\" class=\\"float-right\\">
            New Here?
          </nuxt-link></v-card-text> <v-card-actions class=\\"px-lg-5 px-md-5 px-sm-3\\"><v-checkbox color=\\"primary\\" label=\\"Keep me signed in\\" on-icon=\\"M10,17L5,12L6.41,10.58L10,14.17L17.59,6.58L19,8M19,3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3Z\\" off-icon=\\"M19,3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3M19,5V19H5V5H19Z\\"></v-checkbox> <v-spacer></v-spacer> <v-btn color=\\"primary\\" type=\\"submit\\" large=\\"\\" rounded=\\"\\">
            Sign In
            <v-icon right=\\"\\">
              M10,17V14H3V10H10V7L15,12L10,17M10,2H19A2,2 0 0,1 21,4V20A2,2 0 0,1 19,22H10A2,2 0 0,1 8,20V18H10V20H19V4H10V6H8V4A2,2 0 0,1 10,2Z
            </v-icon></v-btn></v-card-actions></v-form></v-card></v-flex></v-layout>"
`;
